pandas/core/computation/expressions.py

```diff
@@ -167,11 +158,10 @@ def _where_numexpr(cond, a, b):
 
 
 def _has_bool_dtype(x):
+    if isinstance(x, ABCDataFrame):
+        return "bool" in x.dtypes
     try:
-        if isinstance(x, ABCDataFrame):
-            return "bool" in x.dtypes
-        else:
-            return x.dtype == bool
+        return x.dtype == bool
```

Comment:
you can use is_bool_dtype(x) ?

Reply:
could, but thats non-performant, which is a big deal here

Reply:
k